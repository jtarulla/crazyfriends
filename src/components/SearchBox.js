import React from 'react';

const SearchBox = ( { searchChange } ) => {
  return(
    <div className='pa2'>
      <input
        aria-label='Search Friends'
        className='b-green bg-lightest-blue br-pill pa3 ba shadow-5' 
        type='search' 
        placeholder='Search friends' 
        onChange={searchChange}
      /> 
    </div>
  )
}

export default SearchBox