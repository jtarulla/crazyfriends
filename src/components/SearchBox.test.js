import { shallow } from 'enzyme';
import React from 'react';
import SearchBox from './SearchBox';

it('correctly searchs for new friends', () => {
  const searchBox = shallow(<SearchBox />)
  
  expect(searchBox.text()).toEqual('');

  searchBox.find('searchBox').simulate('change');

  expect(searchBox.state()).toEqual({count: 3});

})

// {"children": <input 
//   aria-label="Search Friends" 
//   className="b-green bg-lightest-blue br3 pa3 ba shadow-5" 
//   id="searchBox" 
//   placeholder="Search friends" 
//   type="search" />, 
//   "className": "pa2"
//   }