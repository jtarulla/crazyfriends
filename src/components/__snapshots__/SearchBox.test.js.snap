// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly searchs for new friends 1`] = `ShallowWrapper {}`;
